bots/mailman/bot.js

```diff
@@ -1,44 +1,91 @@
+function locationsSuggestions (params) {
+    var result = {title: "Send location"};
+    var seqArg = params["seq-arg"] ? params["seq-arg"] : "";
+
+    if (seqArg === "Dropped pin")
+    {
+        result.showBack = true;
+        result.height = "max";
+        result.markup = ['view', {},
+                            ['dropped-pin']];
+    }
+    else if (seqArg != "")
+    {
+        result.showBack = true;
+        result.markup = ['scroll-view', {keyboardShouldPersistTaps: "always"},
+                            ['view', {},
+                                ['places-search']]];
+    }
+    else
+    {
+        result.markup = ['scroll-view', {keyboardShouldPersistTaps: "always"},
+                            ['view', {},
+                                ['current-location-map'],
+                                ['current-location'],
+                                ['separator'],
+                                ['places-nearby']]];
+    }
+
+    return result;
+}
+
 status.command({
     name: "location",
-    icon: "location",
     title: I18n.t('location_title'),
     description: I18n.t('location_description'),
-    color: "#a187d5",
     sequentialParams: true,
-    registeredOnly: true,
+    hideSendButton: true,
     params: [{
         name: "address",
         type: status.types.TEXT,
-        placeholder: I18n.t('location_address')
+        placeholder: I18n.t('location_address'),
+        suggestions: locationsSuggestions
     }],
     preview: function (params) {
+        var address = params.address.split("&amp;");
         var text = status.components.text(
             {
                 style: {
-                    marginTop: 5,
+                    marginTop: 0,
                     marginHorizontal: 0,
-                    fontSize: 14,
+                    fontSize: 15,
+                    lineHeight: 23,
                     fontFamily: "font",
                     color: "black"
                 }
-            }, params.address);
-        var uri = "https://maps.googleapis.com/maps/api/staticmap?center="
-            + params.address
-            + "&size=100x100&maptype=roadmap&key=AIzaSyBNsj1qoQEYPb3IllmWMAscuXW0eeuYqAA&language=en"
-            + "&markers=size:mid%7Ccolor:0xff0000%7Clabel:%7C"
-            + params.address;
+            }, address[0]);
+        var uri = "https://api.mapbox.com/styles/v1/mapbox/streets-v10/static/" +
+        address[1] + "," + address[2] + ",10,20" +
+        "/175x58?access_token=pk.eyJ1Ijoic3RhdHVzaW0iLCJhIjoiY2oydmtnZjRrMDA3czMzcW9kemR4N2lxayJ9.Rz8L6xdHBjfO8cR3CDf3Cw";
```

Comment:
don't like that we still have a hardcoded `access_token` here, but I am not sure we can do something with it. please do if we can :)